design-proposals/mass_machine_type_transition.md

```diff
@@ -0,0 +1,299 @@
+and the VM will continue to be tracked until it is restarted and its machine
+type is updated. Both the minimum supported machine type and the target machine
+type will be determined (retrieved) and configured internally when the command is
+invoked, and these values will automatically be updated as future versions are
+released and more machine types are deprecated upon these releases.
+
+If the VM is running, the update will not take effect until it is restarted,
+so the label `restart-vm-required` will be applied to the VM until the VMI is
+removed (indicating the VM has been stopped), upon which the label will be cleared
+after verifying that its machine type has been successfully updated. The job will
+only terminate once all `restart-vm-required` labels have been cleared. By default,
+the user must handle stopping/restarting the labelled VMs manually. This is to allow
+the user to safely update the machine types of their VMs at a time they choose
+without worrying about workload interruptions. However, the user also has the
+option to allow the job to automatically restart every running VM that is being
+updated and apply the changes immediately.
+
+The options to specify by namespace, label-selector, or restart the VMs
+immediately are configurable with the `--namespace`, `--label-selector`,
+and `--restart-now` respectively. In follow-ups to the initial implementation,
+other methods of specifying which VMs to convert the machine types of will be
+added, such as specifying a single VM by name, or configuring a limit on the
+number of VMs that can be restarted at once when restarting the running VMs
+immediately.
+
+Given a VM with the following spec:
+```yaml
+---
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+...
+spec:
+  running: false
+  template:
+    spec:
+      domain:
+        machine:
+          type: pc-q35-rhel8.2.0
+```
+When the MMTT is invoked, the VM spec will be updated to: 
+```yaml
+---
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+...
+spec:
+  running: false
+  template:
+    spec:
+      domain:
+        machine:
+          type: pc-q35-rhel9.2.0
+```
+where `pc-q35-rhel9.2.0` is the latest machine type version.
+
+Along with machine types being in the format `pc-q35-rhelx.x.x`, a VM may have
+a machine type of `q35`, an alias for the current latest machine type *at the time the VM is started*.
+
+For example, this VM:
+```yaml
+---
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+...
+spec:
+  running: true
+  template:
+    spec:
+      domain:
+        machine:
+          type: q35
+```
+may have been started when a RHEL-8 machine type was the latest machine type and
+not been stopped or restarted since. Its VMI would thus look like this:
+```yaml
+---
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+...
+spec:
+  domain:
+    machine:
+      type: q35
+status:
+  machine:
+    type: pc-q35-rhel8.2.0
+```
+and need to have its machine type updated.
```

Comment:
Why do we need to update it if `q35` is floating? Other than migration of course...

Reply:
In this case, the update would take place by simply just restarting the VM (which would automatically end up updating the machine type in the VMI status.machine.type. The machine type in the VM spec would remain `q35` though.

Reply:
The behavior of the `q35` alias is mentioned [here](https://github.com/kubevirt/community/blob/d3938a6d2651014ed197c85d304c7e7bb57b1f6b/design-proposals/mass_machine_type_transition.md?plain=1#L132).

Reply:
I am not quit sure that you mentioned that no action other than restart is required for this alias.